typed-racket-test/succeed/prefab.rkt

```diff
@@ -53,3 +53,45 @@
 ;; Test match (indirectly tests unsafe-struct-ref)
 (match (foo 'x) [(foo s) s])
 (match (foo* 'x) [(foo* s) s])
+
+
+
+(struct msg ([str : String] [maybe-str : (Option String)]) #:prefab)
+
+;; tests let bindings fields of prefab structs
+;; works as expected (requires path-type.rkt to be correct w.r.t. prefabs)
+(: test-let-binding (-> Any String))
+(define (test-let-binding x)
+  (cond
+    [(msg? x)
```

Comment:
@samth thoughts? 

To summarize/recap... currently prefabs have the following properties in TR:

1. There are types which claim to describe them.
2. They can easily expose unsound/unsafe behavior the way they are currently implemented.
3. They are a huge pain to program with (because there are `Path` objects which can refer to their fields, but we are unable to lookup or update the types for those `Path`s (leading to the frustrating behaviors noted in https://github.com/racket/typed-racket/issues/415)).

This PR relieves (3) and does not affect (1) or (2).